client/state/plugins/wporg/selectors.js

```diff
@@ -24,4 +24,36 @@ const isFetched = function( state, pluginSlug ) {
 	return !! plugin.fetched;
 };
 
-export default { getPlugin, isFetching, isFetched };
+const canFetchList = function( state, category, page, searchTerm ) {
+	const { lists } = state.plugins.wporg;
+	if ( searchTerm && lists.fetching.search ) {
+		return false;
+	}
+	if ( lists.fetching[ category ] ) {
+		return false;
+	}
+
+	if ( lists.lastFetchedPage[ category ] >= page ) {
+		return false;
+	}
+
+	return true;
+};
+
+const isFetchingList = function( state, category ) {
+	return !! state.plugins.wporg.lists.fetching[ category ];
+};
+
+const getCurrentSearchTerm = function( state ) {
+	return state.plugins.wporg.lists.currentSearchTerm;
+};
+
+const getList = function( state, category ) {
+	return state.plugins.wporg.lists.fullLists[ category ] ? state.plugins.wporg.lists.fullLists[ category ] : [];
+};
+
+const getShortList = function( state, category ) {
+	return state.plugins.wporg.lists.shortLists[ category ] ? state.plugins.wporg.lists.shortLists[ category ] : [];
+};
+
+export default { getPlugin, isFetching, isFetched, isFetchingList, canFetchList, getList, getShortList, getCurrentSearchTerm };
```

Comment:
let's export each of these in the place they are defined instead of exporting the default export as an object.

```js
export const getShortList = ( state, category ) => get( state, …, [] );
```